iocore/dns/DNS.cc

```diff
@@ -1360,6 +1360,7 @@ dns_process(DNSHandler *handler, HostEnt *buf, int len)
       goto Lerror;
     case SERVFAIL: // recoverable error
       retry = true;
+      break;
```

Comment:
This was the only break I added that does something different.  Please check over this one.